src/js/transfers/nep141~erc20/natural-erc20/sendToNear/index.js

```diff
@@ -97,19 +100,23 @@ export async function initiate ({
 }) {
   // TODO: move to core 'decorate'; get both from contracts
   const sourceTokenName = await getName(erc20Address)
+  // NOTE getDecimals is needed here so it can't be used as a decorator
+  // like getErc20Name
```

Comment:
Why wouldn't we be able to move it to `decorate`? The core library knows how to call the correct library for the given transfer, so it should be able to work properly.

Why call this `naturalDecimals` instead of just `decimals`?

Reply:
- currently the caller of `initiate()` may pass an amount like "2.3" tokens, so initiate must know the correct decimals to apply and record the transfer and make the transaction with the correct decimals (i.e 2.3*10^18). As I understand it `decorate` is only for adding display information for the user. We can remove the `getDecimals` query if `initiate()` already expects a big number string formatted with the correct decimals. Then yes `getDecimals` can be moved to the decorator. But I didn't want to make that change before discussing it.

- natural-erc20 was previously used for erc20s that were born on ethereum vs bridged. So naturalDecimals was to emphasis that the decimals number is always queried from the natural token

Reply:
> We can remove the `getDecimals` query if initiate() already expects a big number string formatted with the correct decimals

I _think_ that seems like the right thing to do

> to emphasize that the decimals number is always queried from the natural token

I don't think we need to expose this subtlety to app developers. Whether stored in localStorage or added by `decorate`, I think just calling the attribute `decimals` seems more expectable. Calling it `naturalDecimals` seems like it adds unnecessary cognitive overhead, imo.

Reply:
got it 👍